linode/framework_provider_config.go

```diff
@@ -0,0 +1,302 @@
+package linode
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/provider"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/logging"
+	"github.com/linode/linodego"
+	"github.com/linode/terraform-provider-linode/linode/helper"
+)
+
+func (fp *FrameworkProvider) Configure(
+	ctx context.Context,
+	req provider.ConfigureRequest,
+	resp *provider.ConfigureResponse,
+) {
+	var data FrameworkProviderModel
+	var meta helper.FrameworkProviderMeta
+
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	fp.HandleDefaults(&data, &resp.Diagnostics)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	fp.InitProvider(&data, req.TerraformVersion, &resp.Diagnostics, &meta)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.ResourceData = &meta
+	resp.DataSourceData = &meta
+	fp.Meta = &meta
+}
+
+// We should replace this with an official validator if
+// HashiCorp decide to implement it in the future
+// feature request track:
+// https://github.com/hashicorp/terraform-plugin-framework-validators/issues/125
+func (fp *FrameworkProvider) ValidateConfig(
+	ctx context.Context,
+	req provider.ValidateConfigRequest,
+	resp *provider.ValidateConfigResponse,
+) {
+	var data FrameworkProviderModel
+
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	_, err := url.Parse(data.APIURL.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Failed to parse the base API URL in the configuration",
+			err.Error(),
+		)
+	}
+}
+
+func GetIntFromEnv(
+	key string,
+	defaultValue basetypes.Int64Value,
+	diags *diag.Diagnostics,
+) basetypes.Int64Value {
+	envVarVal := os.Getenv(key)
+
+	var result basetypes.Int64Value
+	if envVarVal == "" {
+		result = defaultValue
+	} else {
+		intVal, err := strconv.ParseInt(envVarVal, 10, 64)
+
+		if err != nil {
+			diags.AddWarning(
+				fmt.Sprintf(
+					"Failed to parse the environment variable %v "+
+						"to an integer. Will use default value: %v instead",
+					key,
+					defaultValue.ValueInt64(),
+				),
+				err.Error(),
+			)
+			result = defaultValue
+		} else {
+			result = types.Int64Value(intVal)
+		}
+	}
+	return result
+}
+
+func GetStringFromEnv(key string, defaultValue basetypes.StringValue) basetypes.StringValue {
```

Comment:
This can simplify down a bit:
```go
func GetStringFromEnv(key string, defaultValue basetypes.StringValue) basetypes.StringValue {
	envVarVal := os.Getenv(key)

	if envVarVal == "" {
		return defaultValue
	}
	
	return types.StringValue(envVarVal)
}
```